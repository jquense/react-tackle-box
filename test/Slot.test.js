import React from 'react'
import { mount } from 'enzyme'

import createSlot from '../src/Slot'

describe('Slot', () => {
  it('should render to Outlet', async () => {
    const Slot = createSlot()

    const wrapper = mount(
      <div>
        <div className="container">
          inner
          <Slot.Entry>
            <button />
          </Slot.Entry>
        </div>
        outer
        <Slot.Outlet />
      </div>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should throw when trying to render multiple Entries', async () => {
    const Slot = createSlot()
    jest.spyOn(console, 'error').mockImplementation(() => {})

    expect(() =>
      mount(
        <div>
          <Slot.Entry>
            <button />
          </Slot.Entry>
          <Slot.Entry>
            <button />
          </Slot.Entry>
        </div>,
      ),
    ).toThrowError()

    console.error.mockRestore()
  })
})
