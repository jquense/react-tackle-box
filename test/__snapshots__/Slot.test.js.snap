// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slot should render to Outlet 1`] = `"<div><div class=\\"container\\">inner</div>outer<div><button></button></div></div>"`;
